Lib/test/test_marshal.py

```diff
@@ -129,6 +129,32 @@ def test_different_filenames(self):
         self.assertEqual(co1.co_filename, "f1")
         self.assertEqual(co2.co_filename, "f2")
 
+    def test_no_allow_code(self):
+        data = {'a': [({0},)]}
+        dump = marshal.dumps(data, allow_code=False)
+        self.assertEqual(marshal.loads(dump, allow_code=False), data)
+
+        f = io.BytesIO()
+        marshal.dump(data, f, allow_code=False)
+        f.seek(0)
+        self.assertEqual(marshal.load(f, allow_code=False), data)
+
+        co = ExceptionTestCase.test_exceptions.__code__
+        data = {'a': [({co, 0},)]}
+        dump = marshal.dumps(data, allow_code=True)
+        with self.assertRaises(ValueError):
+            marshal.dumps(data, allow_code=False)
+        self.assertEqual(marshal.loads(dump, allow_code=True), data)
```

Comment:
I suggest to move the allow_code=True test before the allow_code=False test (exchange the 2 tests). Same below.